src/Acuminator/Acuminator.Analyzers/Refactorings/EventHandlerSignature/EventHandlerSignatureRefactoring.cs

```diff
@@ -0,0 +1,160 @@
+						&& methodSymbol.Name.EndsWith("_" + eventHandlerInfo.eventType, StringComparison.Ordinal))
+					{
+						(string dacName, string fieldName) = ParseMethodName(methodSymbol.Name, eventHandlerInfo.eventType);
+
+						string title = nameof (Resources.EventHandlerSignatureCodeActionTitle).GetLocalized().ToString();
+						context.RegisterRefactoring(CodeAction.Create(title,
+							ct => ChangeSignatureAsync(context, root, semanticModel, methodNode, methodSymbol,
+								eventHandlerInfo.eventType, genericArgsSymbol, dacName, fieldName, ct), title));
+					}
+				}
+			}
+		}
+
+		private bool IsSuitableForRefactoring(IMethodSymbol symbol, PXContext pxContext)
+		{
+			return symbol?.ContainingType?.OriginalDefinition != null
+			       && symbol.ContainingType.OriginalDefinition.IsPXGraphOrExtension(pxContext)
+			       && symbol.Parameters.Length <= 2
+			       && !symbol.IsOverride
+				   && !symbol.GetAttributes().Any(
+					   attr => attr.AttributeClass != null && attr.AttributeClass.Equals(pxContext.AttributeTypes.PXOverrideAttribute));
+		}
+
+		private Task<Document> ChangeSignatureAsync(CodeRefactoringContext context, 
+			SyntaxNode root, SemanticModel semanticModel,
+			MethodDeclarationSyntax methodDeclaration, IMethodSymbol methodSymbol,
+			EventType eventType, INamedTypeSymbol genericArgsSymbol,
+			string dacName, string fieldName,
+			CancellationToken cancellationToken)
+		{
+			cancellationToken.ThrowIfCancellationRequested();
+
+			var newParameters = methodDeclaration.ParameterList.Parameters;
+			var cacheParameterSymbol = methodSymbol.Parameters[0];
+			ParameterSyntax argsParameter;
+			SyntaxToken parameterName;
+
+			if (eventType == EventType.CacheAttached)
+			{
+				parameterName = Identifier(ArgsParameterName);
+				argsParameter = newParameters[0];
+			}
+			else
+			{
+				newParameters = newParameters.RemoveAt(0); // "PXCache sender" parameter
+				argsParameter = newParameters[0];
+				parameterName = argsParameter.Identifier;
+			}
+
+			newParameters = newParameters.Replace(argsParameter, 
+				CreateArgsParameter(genericArgsSymbol, parameterName, dacName, fieldName));
+
+			var newMethodDeclaration = 
+				ReplaceParameterUsages(methodDeclaration, cacheParameterSymbol,
+					MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, 
+						IdentifierName(parameterName), IdentifierName("Cache")), semanticModel)
+				.WithIdentifier(Identifier(EventHandlerMethodName))
+				.WithParameterList(methodDeclaration.ParameterList.WithParameters(newParameters));
+
+			root = root.ReplaceNode(methodDeclaration, newMethodDeclaration);
+
+			return Task.FromResult(context.Document.WithSyntaxRoot(root));
+		}
+
+		private ParameterSyntax CreateArgsParameter(INamedTypeSymbol genericArgsSymbol, SyntaxToken parameterName, 
+			string dacName, string fieldName)
+		{
+			TypeSyntax identifier;
+
+			if (String.IsNullOrEmpty(fieldName))
```

Comment:
@dropsonic just in case - we have a little extension methods IsNullOrEmpty/IsNullOrWhiteSpace for strings which call corresponding String methods internally, so it could be written like `fieldName.IsNullOrEmpty()`

Reply:
@SENya1990 I prefer not to use them because ReSharper shows "Possible NullReferenceException" warning in that case.

Reply:
@dropsonic we can add JetBrains Annotation attributes to them to fix the situation.

Reply:
@SENya1990 I couldn't find any annotation that tells ReSharper that this method performs a null checking.